### DISCUSSION TITLE
Possible to hide tqdm-like progress bar updates during JupyterBook build?

### DISCUSSION THREAD

**Discussion by matthewfeickert** [executablebooks/meta#1067] [Q&A]
Hi. If you have a `tqdm` progress bar for a piece of code that gets executed in a notebook during a JupyterBook build the updates of the progress bar get frozen into the build

Example from local build of https://github.com/pyhf/pyhf-tutorial:

![image](https://github.com/executablebooks/meta/assets/5142394/52a8350b-e25f-4d59-8e95-c815341964ea)

Is it possible to have JupyterBook know how to ignore the updates or to only keep the final update? I assume no, but I thought I would ask.

Related: https://github.com/pyhf/pyhf-tutorial/pull/66

**Comment by agoose77** [comment]
Hmm, my initial thoughts here are that we probably need to explicitly handle the carriage return. That could be an option. You could also keep the output by placing only the `infer.hypotest` call in a separate cell that has the `scroll-output` or `hide-output` tag. Then, it's effectively hidden.

**Reply by matthewfeickert** [reply]
Hm. I've added tags in Jupyter Lab using the [JupyterLab Cell Tags plugin](https://jupyterbook.org/en/stable/content/metadata.html#adding-tags-using-notebook-interfaces)


```console
$ git grep "scroll-output"
book/Toys.ipynb:     "\"scroll-output\""
book/Toys.ipynb:     "\"scroll-output\""
book/Toys.ipynb:     "\"scroll-output\""
book/UsingCalculators.ipynb:     "\"scroll-output\""
$ git diff origin/main -- book/Toys.ipynb
...
@@ -131,7 +130,15 @@
   {
    "cell_type": "code",
    "execution_count": null,
-   "metadata": {},
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": [
+     "\"scroll-output\""
+    ]
+   },
...
```

but I'm still seeing the issue above. We still want the final output of the cell to be shown, just not the updates.

**Reply by agoose77** [reply]
Luckily, recent JupyterLab's shouldn't need any extension to add tags
![image](https://github.com/executablebooks/meta/assets/1248413/de87271c-ffb9-44b1-84db-b78519c742fa)

> We still want the final output of the cell to be shown, just not the updates.

Yep, this setting won't handle the problem of `\r` carriage returns. I would imagine that a post-process operation would be required to deal with these. I wonder what JupyterLab does. Oh, [here it is](https://github.com/jupyterlab/jupyterlab/blob/ac8f9e2f47b271785127757541a6fd8aa10c4d31/packages/outputarea/src/model.ts#L500-L513) — a post-process operation!
